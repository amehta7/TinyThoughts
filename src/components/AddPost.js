import React, { memo } from 'react'
import Input from './Input'
import Text from './Text'
import Select from './Select'
import { connect } from 'react-redux'
import { addPost } from '../store/actions/index'

const AddPost = memo(() => {
  return (
    <div className='add-post'>
      <form>
        <h2>New Post</h2>
        <div className='add-post-form'>
          <Input label='Title' type='text' hasError={false} />
          <Text
            label='Type in your thoughts in 200 words or less'
            type='text'
            hasError={false}
            maxLength={200}
          />
          <div className='char-count'>of 200 characters</div>
          <Select label='Select a category' type='text' hasError={false} />
          <button id='add-post-btn'>+ Add</button>
        </div>
      </form>
    </div>
  )
})

const mapStateToProps = (state) => {
  return {
    posts: state.posts.posts,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onAddPost: (post) => dispatch(addPost(post)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AddPost)
