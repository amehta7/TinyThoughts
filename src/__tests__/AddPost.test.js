import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import AddPost from "../components/AddPost";

const mockStore = configureStore([thunk]);

describe("AddPost component", () => {
  it("Renders the AddPost component successfully", () => {
    const store = mockStore({
      users: {
        user: { id: 1, name: "Joe Mockery" }
      },
      posts: {
        categories: [
          { id: 1, title: "JavaScript" },
          { id: 2, title: "Python" }
        ]
      },
      errors: {
        error: false
      }
    });
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <AddPost />
        </MemoryRouter>
      </Provider>
    );

    return expect(wrapper.find(AddPost)).toMatchSnapshot();
  });

  it("Local state, event listeners and controlled components work as desired", async () => {
    const store = mockStore({
      users: {
        user: { id: 1, name: "Joe Mockery" }
      },
      posts: {
        categories: [
          { id: 1, title: "JavaScript" },
          { id: 2, title: "Python" }
        ]
      },
      errors: {
        error: false
      }
    });

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <AddPost />
        </MemoryRouter>
      </Provider>
    );

    await wrapper.find("input").simulate("change", {
      target: {
        value: "Demo post"
      }
    });

    await wrapper.find("textarea").simulate("change", {
      target: {
        value: "This is the content of the post"
      }
    });

    await wrapper.find("Select").simulate("change", {
      target: {
        value: 2
      }
    });

    await wrapper.find("button").simulate("click");

    expect(wrapper.find("Text").prop("value")).toEqual(
      "This is the content of the post"
    );

    expect(wrapper.find("Input").prop("value")).toEqual("Demo post");

    expect(wrapper.find(".char-count").text()).toEqual("31 of 200 Characters");
  });
});
