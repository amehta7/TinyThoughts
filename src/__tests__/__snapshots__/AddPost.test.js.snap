// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPost component Renders the AddPost component successfully 1`] = `
<Memo()>
  <div
    className="add-post"
  >
    <h2>
      New Post
    </h2>
    <div
      className="add-post-form"
    >
      <Input
        hasError={false}
        label="Title"
        onInput={[Function]}
        type="text"
        value=""
      >
        <div
          className="input-box"
        >
          <span
            className="label"
          >
            Title
          </span>
          <input
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </Input>
      <Text
        hasError={false}
        label="Type in your thoughts in 200 characters or less"
        maxLength={200}
        onUpdate={[Function]}
        value=""
      >
        <div
          className="text-box"
        >
          <span
            className="label"
          >
            Type in your thoughts in 200 characters or less
          </span>
          <textarea
            maxLength={200}
            onChange={[Function]}
            value=""
          />
        </div>
      </Text>
      <div
        className="char-count"
      >
        0
         of 200 Characters
      </div>
      <Select
        hasError={false}
        label="Select a category"
        onSet={[Function]}
        options={
          Array [
            Object {
              "id": 1,
              "title": "JavaScript",
            },
            Object {
              "id": 2,
              "title": "Python",
            },
          ]
        }
      >
        <div
          className="select-list"
        >
          <span
            className="label"
          >
            Select a category
          </span>
          <select
            defaultValue="Select"
            onChange={[Function]}
          >
            <option
              disabled={true}
            >
              Select
            </option>
            <option
              key="1"
              value={1}
            >
              JavaScript
            </option>
            <option
              key="2"
              value={2}
            >
              Python
            </option>
          </select>
        </div>
      </Select>
      <button
        id="add-post-btn"
        onClick={[Function]}
      >
        + Add
      </button>
    </div>
  </div>
</Memo()>
`;
